frappe/public/js/frappe/ui/group_by/group_by.js

```diff
@@ -51,21 +51,29 @@ frappe.ui.GroupBy = class {
 	}
 
 	show_hide_aggregate_on() {
-		this.report_view.meta.fields.forEach((field) => {
-			let fn = this.aggregate_function_select.val();
-			if(fn === 'sum' || fn === 'avg') {
-				// pick numeric fields for sum / avg
-				if(frappe.model.is_numeric_field(field.fieldtype)) {
-					this.aggregate_on_select.append(
-						$('<option>', { value : field.fieldname })
-							.text(field.label));
+
+		for (let doctype in this.all_fields) {
+			const doctype_fields = this.all_fields[doctype];
+			doctype_fields.forEach(field => {
+				let fn = this.aggregate_function_select.val();
+				if(fn === 'sum' || fn === 'avg') {
+					// pick numeric fields for sum / avg
+					if(frappe.model.is_numeric_field(field.fieldtype)) {
+						let option_text = doctype == this.doctype
+							? field.label
+							: `${field.label} (${doctype})`;
+						this.aggregate_on_select.append(
```

Comment:
This can append duplicate options on changing aggregate function
![duplicate_options](https://user-images.githubusercontent.com/13928957/75686273-fcec2780-5cc1-11ea-9cd0-139dd639e469.gif)